### PR TITLE
Tidy up the State and Backends page

### PR DESCRIPTION
**Pull request by cnunciato**
Noticed a few comments on this page that had largely been addressed, but went ahead and cleaned up a little anyway:

* Added choosers for cloud providers
* Made bucket-name references more consistent
* Reordered a couple of sections (e.g., moving the Backend Info Check section below the set of all backend options)
* Adjusted a couple of headings and their placement in the TOC hierarchy

Part of #4064.

### CODE REVIEW

File: content/docs/intro/concepts/state.md [hunk 1]
```diff
@@ -47,31 +47,48 @@ security for most users.
...
+
+To use the [cloud storage or remote](#cloud-storage-or-remote) option, run `pulumi login
+--cloud-url <your-cloud-storage-url>` to have Pulumi store your checkpoint files using a
+supported cloud provider like AWS, Azure, or Google Cloud Platform. Checkpoint files are
+also stored in `.pulumi` directory. For example, if you were using the Amazon S3
+self-managed backend, your checkpoint files would be stored at
+`s3://<my-pulumi-state-bucket>/.pulumi` where `<my-pulumi-state-bucket>` represents the
+name of your S3 bucket.
+
+The following links offer additional details for setting up your provider-specific
```
Comment (spara) [inline content/docs/intro/concepts/state.md:64]:
```suggestion
The following links offer additional details for setting up provider-specific
```

File: content/docs/intro/concepts/state.md [hunk 2]
```diff
@@ -207,11 +207,11 @@ file://~`.
 #### Cloud storage or remote
 
 To use a remote backend instead with your preferred cloud storage provider,
-simply pass the `--cloud-url` or `-c` flag and your storage bucket or blob URL
+simply pass the `--cloud-url` (or `-c`) flag and your storage bucket or blob URL
```
Comment (spara) [inline content/docs/intro/concepts/state.md:210]:
```suggestion
pass the `--cloud-url` (or `-c`) flag and your storage bucket or blob URL
```

**Review by spara — COMMENTED** [review]
just a few nits, otherwise lgtm

**Comment by cnunciato** [comment]
@spara suggestions applied. Thanks!

**Pull request merged**